dbt/scripts/utils/export.py

```diff
@@ -101,149 +173,190 @@ def export_models(
-    )
+        cursor_class=pyathena.pandas.cursor.PandasCursor,
+        # Unload query results to speed up execution times
+    ).cursor(unload=True)
 
+    models_for_export: list[ModelForExport] = []
     for model in models:
         # Extract useful model metadata from the columns we queried in
         # the `dbt list` call above
         model_name = model["name"]
+        tags = model["tags"]
         relation_name = model["relation_name"]
         export_name = model["config"]["meta"].get("export_name") or model_name
         template = model["config"]["meta"].get("export_template") or model_name
 
         # Define inputs and outputs for export based on model metadata
         template_path = os.path.join("export", "templates", f"{template}.xlsx")
-        template_exists = os.path.isfile(template_path)
-        output_path = os.path.join("export", "output", f"{export_name}.xlsx")
 
         print(f"Querying data for model {model_name}")
         query = f"SELECT * FROM {relation_name}"
         if where:
             query += f" WHERE {where}"
         print(f"> {query}")
-        model_df = pd.read_sql(query, conn)
+        model_df = cursor.execute(query).as_pandas()
 
-        # Delete the output file if one already exists
-        pathlib.Path(output_path).unlink(missing_ok=True)
+        models_for_export.append(
+            ModelForExport(
+                name=model_name,
+                tags=tags,
+                df=model_df,
+                export_format=(
+                    model["config"]["meta"].get("export_format", {})
+                ),
+                export_filename=f"{export_name}.xlsx",
+                template_path=template_path,
+            )
+        )
 
-        if template_exists:
-            print(f"Using template file at {template_path}")
-            shutil.copyfile(template_path, output_path)
-        else:
-            print("No template file exists; creating a workbook from scratch")
+    return models_for_export
+
+
+def save_model_to_workbook(
+    model: ModelForExport,
+    output_dir: str | None = None,
+) -> pathlib.Path:
+    """
+    Given a ModelForExport object containing row-level data and table-level
+    metadata for a model, process that model and save its data to the
+    configured location on disk.
+
+    Arguments:
+
+        * model_for_export (ModelForExport): Object containing row-level model
+            data and table-level metadata about models that should be saved
+        * output_dir (str): Optional Unix path to directory where output files
+            should be stored
+
+    Returns a pathlib.Path object representing the path to the file that the
+    function created.
+    """
+    # Compute the full filepath for the output
+    output_dirpath = (
+        pathlib.Path(output_dir)
+        if output_dir
+        else pathlib.Path("export/output")
+    )
+    output_path = output_dirpath / model.export_filename
```

Comment:
Beyond this step, very little of the logic of this function has changed, it's just present in the diff because we've extracted it out from the old `export_models()` implementation. I'll leave a comment on anything major that needs attention.